File: toms-allergies/toms-allergies.py
class Allergies(object):
    def __init__(self,n):
        self.n = bin(n%256)[2:]
        
    def allergies(self):
        lista = ['eggs', 'peanuts', 'shellfish', 'strawberries', 'tomatoes', 'chocolate', 'pollen', 'cats']
        return sorted([lista[i] for i,j in enumerate(self.n[::-1]) if j == "1"])
    
    def is_allergic_to(self, m):
        return m in self.allergies()

File: toms-allergies/toms-allergies.js
function Allergies(score){
   if(typeof score!= "number"||score==null||score % 1 !== 0){throw new TypeError()}

  this.ALLERGY_SCORES = ["eggs","peanuts","shellfish","strawberries","tomatoes","chocolate","pollen","cats"]
    
  this.isAllergicTo=function(allergen){
    return this.allergies(score).join().indexOf(allergen)==-1? false:true}
      
  this.allergies=function(){
    var t = []
    var j = (score%256).toString(2).split("").reverse().join("")
    for(var i=0;i<j.length;i++){
    if(j[i]==1){ t.push(this.ALLERGY_SCORES[i]) }   }  
    return t.sort()
  }
}